app/src/main/java/crux/bphc/cms/helper/CourseDataHandler.java

```diff
@@ -79,8 +79,10 @@ public String getActionBarTitle(int courseId){
             if (realm.where(Course.class).equalTo("id", course.getId()).findFirst() == null) {
                 newCourses.add(course);
             }
+            else
+                course.setFavoriteStatus(realm.where(Course.class).equalTo("id", course.getId()).findFirst().isFavorite());
         }
-
+        rearrangeCourses(courseList);
```

Comment:
Is it really required to sort the courses before committing them to persistent storage?

Reply:
Without this function call, the favourite courses are moved from the top to their original position on the MyCourses list, as soon as the screen is refreshed.

Reply:
Even in that case, it is the responsibility of the view and/or controller to order the data in the way fit for consumption by the view. Doing such ordering before saving it is to persistent storage isn't needed.